### PR TITLE
[Keystone][Deployments] Make adding NOPs and DONs idempotent

### PR DESCRIPTION
**Pull request by bolekk**
Similarly to adding capabilities or nodes, don't add NOPs or DONs that already exist.

### CODE REVIEW

File: deployment/keystone/deploy.go [hunk 1]
```diff
@@ -421,8 +420,36 @@ type RegisterNOPSResponse struct {
...
-func RegisterNOPS(ctx context.Context, req RegisterNOPSRequest) (*RegisterNOPSResponse, error) {
-	nops := req.Nops
+func RegisterNOPS(ctx context.Context, lggr logger.Logger, req RegisterNOPSRequest) (*RegisterNOPSResponse, error) {
+	existingNops, err := req.Registry.GetNodeOperators(&bind.CallOpts{})
+	if err != nil {
+		return nil, err
+	}
+	existingNopsAddrToID := make(map[string]uint32)
+	for id, nop := range existingNops {
+		existingNopsAddrToID[nop.Admin.String()] = uint32(id)
```
Comment (krehermann) [inline deployment/keystone/deploy.go]: the admin strings are phony in some of our deployments and not unique.

maybe hash the name,addr and use that as the key

**Review by krehermann — APPROVED** [review]
admin addr not unique in some existing contracts. this will change won't behave as expected; simplest fix is to hash the name and admin addr as map key.

there is nothing in the contract guaranteeing unique of admin addrs.

**Review by krehermann — CHANGES_REQUESTED** [review]
meant to request change not approve

File: deployment/keystone/deploy.go (hunk 1 above)
Comment (krehermann) [reply deployment/keystone/deploy.go]: i think you can just use the struct as the key

**Review by krehermann — APPROVED** [review]

**Review by patrickhuie19 — APPROVED** [review]

**Pull request merged**